packages/react-native/React/Views/RCTViewManager.m

```diff
@@ -292,34 +291,45 @@ - (RCTShadowView *)shadowView
 
 RCT_CUSTOM_VIEW_PROPERTY(accessibilityRole, UIAccessibilityTraits, RCTView)
 {
-  #if !TARGET_OS_OSX // [macOS]
+#if !TARGET_OS_OSX // [macOS]
   UIAccessibilityTraits accessibilityRoleTraits =
       json ? [RCTConvert UIAccessibilityTraits:json] : UIAccessibilityTraitNone;
   if (view.reactAccessibilityElement.accessibilityRoleTraits != accessibilityRoleTraits) {
     view.accessibilityRoleTraits = accessibilityRoleTraits;
     view.reactAccessibilityElement.accessibilityRole = json ? [RCTConvert NSString:json] : nil;
     [self updateAccessibilityTraitsForRole:view withDefaultView:defaultView];
   }
-  #else // [macOS
-    if (json) {
-      view.reactAccessibilityElement.accessibilityRole = [RCTConvert accessibilityRoleFromTraits:json];
-    } else {
-      view.reactAccessibilityElement.accessibilityRole = defaultView.accessibilityRole;
-    }
-  #endif // macOS]
+#else // [macOS
+  // accessibilityRoleInternal is used to cache the converted value from the prop
+  view.reactAccessibilityElement.accessibilityRoleInternal = json ? [RCTConvert accessibilityRoleFromTraits:json withAriaMappings:NO] : nil;
```

Comment:
Should we cache in a local variable instead?

Reply:
I'm not entirely sure what you mean. Local to this function? Caching ViewManager wouldn't make sense, right? That's one-to-many (RCTViewManager to UIViews). We cache on the actual UIView so that regardless of the order of operations between setting `accessibilityRole` and `role` `updateAccessibilityRole` has access to both converted values and can correctly apply the precedence we want. I know it's kind of an edge case for users to be setting/unsetting both props, but that's how Meta implemented and spec'd the behavior of these props.

Prior to this change, accessibilityRoleInternal and role already existed but were never actually used.  This updates them to be used in a fashion that parallels the iOS implementation Meta does.

Reply:
Ah, ok, we're handling the edge case of when things are set in a different order. Fabric makes it easier because views have a reference to all their props, and you just setall the props from the property bag and then do any "final" logic after.